test/jdk/java/nio/channels/Channels/TransferTo.java

```diff
@@ -138,27 +138,38 @@ public void testStreamContents(InputStreamProvider inputStreamProvider,
      */
     @Test
     public void testMoreThanTwoGB() throws IOException {
-        // preparing two temporary files which will be compared at the end of the test
-        Path sourceFile = Files.createTempFile(null, null);
-        Path targetFile = Files.createTempFile(null, null);
-
-        // writing 3 GB of random bytes into source file
-        for (int i = 0; i < NUM_WRITES; i++)
-            Files.write(sourceFile, createRandomBytes(BYTES_PER_WRITE, 0), StandardOpenOption.APPEND);
-
-        // performing actual transfer, effectively by multiple invocations of Filechannel.transferTo(FileChannel)
-        long count;
-        try (InputStream inputStream = Channels.newInputStream(FileChannel.open(sourceFile));
-                OutputStream outputStream = Channels
-                        .newOutputStream(FileChannel.open(targetFile, StandardOpenOption.WRITE))) {
-            count = inputStream.transferTo(outputStream);
+        Path sourceFile = null;
+        Path targetFile = null;
+        try {
+            // preparing two temporary files which will be compared at the end of the test
+             sourceFile = Files.createTempFile("test2GBSource", null);
```

Comment:
If this throws then Files.delete(sourceFile) will throw an NPE. I think it would be just a bit more robust to initialize sourceFile to Files.createTempFile(...) rather than null.